client/state/terms/test/selectors.js

```diff
@@ -134,6 +141,222 @@ describe( 'selectors', () => {
 		} );
 	} );
 
+	describe( 'getTermsForQueryIgnoringPage', () => {
+		beforeEach( () => {
+			getTermsForQuery.memoizedSelector.cache.clear();
+		} );
+
+		it( 'should return null if last page is not known', () => {
+			const terms = getTermsForQueryIgnoringPage( {
+				terms: {
+					queriesLastPage: {}
+				}
+			}, 2916284, 'category', { search: 'Hello' } );
+
+			expect( terms ).to.be.null;
+		} );
+
+		it( 'should return terms ignoring page param', () => {
+			const terms = getTermsForQueryIgnoringPage( {
+				terms: {
+					queriesLastPage: {
+						2916284: {
+							category: {
+								'{"search":"hello"}': 2
+							}
+						}
+					},
+					queries: {
+						2916284: {
+							category: {
+								'{"search":"hello"}': [ 123 ],
+								'{"search":"hello","page":2}': [ 124 ]
+							}
+						}
+					},
+					items: {
+						2916284: {
+							category: {
+								123: {
+									ID: 123,
+									name: 'Chicken',
+									slug: 'chicken'
+								},
+								124: {
+									ID: 124,
+									name: 'Ribs',
+									slug: 'ribs'
+								}
+							}
+						}
+					}
+				}
+			}, 2916284, 'category', { search: 'Hello', page: 2 } );
+
+			expect( terms ).to.eql( [ {
+				ID: 123,
+				name: 'Chicken',
+				slug: 'chicken'
+			}, {
+				ID: 124,
+				name: 'Ribs',
+				slug: 'ribs'
+			} ] );
+		} );
+	} );
+
+	describe( 'getTermsLastPageForQuery()', () => {
+		it( 'should return null if the terms query is not tracked', () => {
+			const lastPage = getTermsLastPageForQuery( {
+				terms: {
+					queriesLastPage: {}
+				}
+			}, 2916284, 'category', { search: 'Hello' } );
+
+			expect( lastPage ).to.be.null;
+		} );
+
+		it( 'should return the last page value for a query', () => {
+			const lastPage = getTermsLastPageForQuery( {
+				terms: {
+					queriesLastPage: {
+						2916284: {
+							category: {
+								'{"search":"hello"}': 4
+							}
+						}
+					}
+				}
+			}, 2916284, 'category', { search: 'Hello' } );
+
+			expect( lastPage ).to.equal( 4 );
+		} );
+
+		it( 'should return the last page value for a terms query, even if including page param', () => {
+			const lastPage = getTermsLastPageForQuery( {
+				terms: {
+					queriesLastPage: {
+						2916284: {
+							category: {
+								'{"search":"hello"}': 4
+							}
+						}
+					}
+				}
+			}, 2916284, 'category', { search: 'Hello', page: 2 } );
+
+			expect( lastPage ).to.equal( 4 );
+		} );
+	} );
+
+	describe( 'getTermsHierarchyForQueryIgnoringPage()', () => {
+		beforeEach( () => {
+			getTermsHierarchyForQueryIgnoringPage.memoizedSelector.cache.clear();
```

Comment:
Might be easier/safer to include all of the memoized selector cache clearing in a single `beforeEach` block at the top-level `describe`